website/project/views/contributor.py

```diff
@@ -669,6 +670,12 @@ def claim_user_form(auth, **kwargs):
     user.update_guessed_names()
     # The email can be the original referrer email if no claimer email has been specified.
     claimer_email = unclaimed_record.get('claimer_email') or unclaimed_record.get('email')
+
+    # If there is a registered user with this email, redirect to 're-enter password' page
+    user_from_email = User.find_one(Q('username', 'eq', claimer_email))
```

Comment:
@brianjgeiger @caseyrollins thanks a lot, totally my fault, I had thought https://github.com/CenterForOpenScience/osf.io/commit/b885376a04cd40ad995716ea04536a06d1819350 was the more edgy commit without realizing it was an old one.